### PR TITLE
feat(bundler): Add Debian package support for updater bundles

### PR DESCRIPTION
**Pull request by jLynx**
## Description

This PR adds support for Debian (.deb) packages in the updater bundle system, allowing Debian packages to be included in the self-contained updater functionality alongside existing formats like AppImage, NSIS, and Windows MSI.

_Note: This PR relies on https://github.com/tauri-apps/plugins-workspace/pull/1991_

## Changes

- Added `PackageType::Deb` to the list of self-contained updater package types
- Modified the Linux bundler to handle both AppImage and Debian package formats
- Extended the updater bundle functionality to create tar.gz archives for Debian packages
- Updated signing functionality to include Debian packages
- Fixed JSON formatting in schema documentation examples

## Testing

I have tested the following scenarios:

1. Building a Debian package with updater enabled
2. Verifying the .deb archive creation for Debian packages
3. Confirming the signing process works correctly for Debian packages
4. Ensuring existing functionality for other package types remains unchanged

### CODE REVIEW

**Comment by jLynx** [comment]
Removed support for .deb.tar.gz as requested by @FabianLars https://discord.com/channels/616186924390023171/1301285233991356506/1302596935101976616

**Review by amrbashir — COMMENTED** [review]
Could you also add a change file in .changes directory?

File: .changes/deb-updater-support.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@tauri-apps/cli': minor:feat
+'@tauri-apps/bundler': minor:feat
```
Comment (amrbashir) [inline .changes/deb-updater-support.md]:
```suggestion
'tauri-bundler': minor:feat
'tauri-cli': minor:feat
'@tauri-apps/cli': minor:feat
```

File: .changes/deb-updater-support.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+---
+
+Add support for Debian (`.deb`) packages in the updater bundle system, enabling self-contained updater functionality for Debian packages alongside existing formats like AppImage, NSIS, and Windows MSI.
```
Comment (amrbashir) [inline .changes/deb-updater-support.md]:
```suggestion
Generate signature for `.deb` packages when `createUpdaterArtifacts` option is enabled.
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by amrbashir — APPROVED** [review]

**Pull request merged**